src/lib/ipsec/esp.lua

```diff
@@ -168,18 +161,72 @@ function esp_v6_decrypt:decapsulate (p)
                    "flow_id=" .. tostring(self.ip:flow_label()) .. ", " ..
                    "reason='" .. reason .. "'";
       logger:log("Rejecting packet ("..info..")")
-      return false
+      -- RFC4303 is somewhat unclear on a) whether or not packets that look replayed are eligible to trigger
+      -- recyns, and b) what 'consecutive' means in "number of consecutive packets that fail authentication":
+      -- Consecutive in time? Consecutive in their seq_low's?
+      -- Assuming a) should be 'yes' since the to-be-resynced packets might have seq_low's that just happen
+      -- to fall inside the replay window (seq_lo-wise) and would look replayed, yet aren't.
+      -- Assuming b) means 'in time', since systematic loss could stall resync indefinitely.
+      self.decap_fail = self.decap_fail + 1
+      if self.decap_fail >= self.resync_threshold then
+         local resync_start
+         if seq_high >= 0 then -- We failed to decrypt in-place, undo the damage to recover the original ctext
+            gcm:encrypt(ctext_start, iv_start, seq_low, seq_high, ctext_start, ctext_length, ffi.new("uint8_t[?]", gcm.AUTH_SIZE))
+            resync_start = seq_high + 1
+         else
+            resync_start = self:seq_high() + 1 -- use the last seq_high we saw if it looked replayed
+         end
+         self.decap_fail = 0 -- avoid immediate re-triggering if resync fails
+         local seq_high_resynced = self:resync(p, seq_low, resync_start, self.resync_attempts)
+         if seq_high_resynced >= 0 then
+            seq_high = seq_high_resynced
+            -- resynced! the data has been decrypted in the process so we're ready to go
```

Comment:
That is being taken care of by the call to track_seq_no a few lines below this. Effectively it calls [advance_window](https://github.com/vanfstd/snabb/blob/ef2b4027d3baaabf337e8fb521d3bdd76f994b15/src/lib/ipsec/track_seq_no.c#L44) with the resynced-on sequence number, resulting in `diff` in that function to be large so the entire window is cleared.
Might still make sense to add a test for this, though.